src/provider/states/pod/creating_config.rs

```diff
@@ -364,11 +364,20 @@ impl State<PodState> for CreatingConfig {
             );
         };
 
-        // Write the config files
-        let config_dir = pod_state.get_service_config_directory();
         for (target_path, volume) in volume_mounts {
+            // This is a hack for the NiFi operator. We need the volume mounts for NiFi to point to
```

Comment:
I think we could make this a little bit less hacky by actually using the template renderer instead of string substitution here.
The current implementation would miss `{{ packageroot }}` for example, which is still a valid Handlebars reference.

Something along these lines should have a similar effect: 

```rust
 let joined_target_path = pod_state.get_service_package_directory().join(
                CreatingConfig::render_config_template(&template_data, &target_path).unwrap(),
            );
```

{{packageroot}}/tmp" should be rendered to for example "/opt/stackable/packages/kafka-2.6.0/tmp"

Since PathBuf::join() resolves absolute paths absolutely this would lead to 
"/opt/stackable/servicedirectory".join("/opt/stackable/packages/kafka-2.6.0/tmp") -> "/opt/stackable/packages/kafka-2.6.0/tmp"

Reply:
Yes thank you, that looks much better :)